xbmc/pvr/PVRDatabase.cpp

```diff
@@ -636,6 +636,18 @@ bool CPVRDatabase::Persist(CPVRChannelGroup &group)
   {
     CSingleLock lock(group.m_critSection);
 
+    /* Prevent AutoIncValue usage for internal groups as long they have fixed value */
+    if (group.GroupID() <= 0 && group.IsInternalGroup())
```

Comment:
this condition will never be valid because a internal group never holds a GroupID lower 0. Take a look a the ctor of `CPVRChannelGroupInternal`.